model/event/event_finder.go

```diff
@@ -61,6 +63,36 @@ func CountProcessEvents(taskID string) (int, error) {
 	return db.CountQ(TaskLogCollection, db.Query(filter))
 }
 
+func FindLastProcessedEvent() (*EventLogEntry, error) {
+	q := db.Query(bson.M{
+		processedAtKey: bson.M{
+			"$exists": true,
+		},
+	}).Sort([]string{"-" + processedAtKey}).Limit(1)
+
+	e := EventLogEntry{}
+	if err := db.FindOneQ(AllLogCollection, q, &e); err != nil {
+		if err == mgo.ErrNotFound {
+			return nil, nil
+		}
+		return nil, errors.Wrap(err, "failed to fetch most recently processed event")
+	}
+
+	return &e, nil
+}
+
+func CountUnprocessedEvents() (int, error) {
+	q := db.Query(UnprocessedEvents())
+	q.Limit(100000)
```

Comment:
Why do you need to set a limit on this?

Reply:
The count query takes a surprisingly long amount of time. In the event of a backlog of events, the limit is to prevent this query from taking too long.

I will add a docstring with this info

Reply:
should we think about indexing it differently?

Reply:
@tychoish Index is {processed_at: 1}. Query: {processed_at: {$exists: false}}. Can you recommend a better index or query for this?

Reply:
db.CountQ doesn't seem to propogate the limit. 

does a range query work better for the index? It's worth doing an explain.